build: Fix qt build for riscv64

hebasto opened the pull request:
Fixes #29.

#### Guix build:
```
$ env HOSTS='riscv64-linux-gnu' ./contrib/guix/guix-build
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
9399101a4268eb78b4c13d0895964862211d87e9a12bd154c11ea4ca257f02d6  guix-build-3a0f948c0bc9/output/dist-archive/bitcoin-3a0f948c0bc9.tar.gz
c4c57780ee1d99721dc370c35cc68dfccdd5455a29c0c526a1171f3b8dfa4dc8  guix-build-3a0f948c0bc9/output/riscv64-linux-gnu/SHA256SUMS.part
4840f8a57c43e3956073681aeb3779f326fa89102a643c5bf5acd26790107284  guix-build-3a0f948c0bc9/output/riscv64-linux-gnu/bitcoin-3a0f948c0bc9-riscv64-linux-gnu-debug.tar.gz
95910c5070e14532c2b96a10662c1ae77c45199649234248ee71017e15cc7a71  guix-build-3a0f948c0bc9/output/riscv64-linux-gnu/bitcoin-3a0f948c0bc9-riscv64-linux-gnu.tar.gz
```

jarolrod commented on depends/patches/qt/fix_riscv_atomic.patch:
```diff
@@ -0,0 +1,17 @@
+A workaroud the error when build for riscv64.
```
```suggestion
A workaround for error when building for riscv64.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/30#issuecomment-908117764).

jarolrod commented:
GUIX hashes mine match @hebasto:
``` diff
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

00bd2214012476a91e32cddc720ef3ba12dcd992c0fea4f656f8e1ca672b1768  guix-build-bd86bb536645/output/dist-archive/bitcoin-bd86bb536645.tar.gz
1f2f80002f6bda09073f3da6b292d232e1c657ecf1e6f0c59ac01b1f5d3dd474  guix-build-bd86bb536645/output/x86_64-w64-mingw32/SHA256SUMS.part
b988b0ce195f536799b30194f573a0ae0d26a32b00c1a5b54039d62e60f92db9  guix-build-bd86bb536645/output/x86_64-w64-mingw32/bitcoin-bd86bb536645-win-unsigned.tar.gz
e89793b517eb171ecf0668f5f004af2df43077614f2cb22e0eb9af736b51dc80  guix-build-bd86bb536645/output/x86_64-w64-mingw32/bitcoin-bd86bb536645-win64-debug.zip
ccc6fc78cf0de61772bab59a83b2370aa825bd5c6d0af126b19c02f985b04fda  guix-build-bd86bb536645/output/x86_64-w64-mingw32/bitcoin-bd86bb536645-win64-setup-unsigned.exe
d753a67e614ad33d6c07e8e32183feec81dee0edd51f319e9bea698c985cf0b6  guix-build-bd86bb536645/output/x86_64-w64-mingw32/bitcoin-bd86bb536645-win64.zip
```

hebasto commented:
Updated 5ed24d52e3a9c085c9cbeb9b4b43bf3bf5da7422 -> 3a0f948c0bc93b571168165c54966bf164dc356a ([pr30.02](https://github.com/hebasto/gui-qml/commits/pr30.02) -> [pr30.03](https://github.com/hebasto/gui-qml/commits/pr30.03)):
- rebased on top of the #28 
- addressed @jarolrod's [comment](https://github.com/bitcoin-core/gui-qml/pull/30#discussion_r698061571)

jarolrod commented:
GUIX hashes mine match @hebasto:

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

9399101a4268eb78b4c13d0895964862211d87e9a12bd154c11ea4ca257f02d6  guix-build-3a0f948c0bc9/output/dist-archive/bitcoin-3a0f948c0bc9.tar.gz
c4c57780ee1d99721dc370c35cc68dfccdd5455a29c0c526a1171f3b8dfa4dc8  guix-build-3a0f948c0bc9/output/riscv64-linux-gnu/SHA256SUMS.part
4840f8a57c43e3956073681aeb3779f326fa89102a643c5bf5acd26790107284  guix-build-3a0f948c0bc9/output/riscv64-linux-gnu/bitcoin-3a0f948c0bc9-riscv64-linux-gnu-debug.tar.gz
95910c5070e14532c2b96a10662c1ae77c45199649234248ee71017e15cc7a71  guix-build-3a0f948c0bc9/output/riscv64-linux-gnu/bitcoin-3a0f948c0bc9-riscv64-linux-gnu.tar.gz
```

jarolrod commented on depends/patches/qt/fix_riscv_atomic.patch:
```diff
@@ -0,0 +1,17 @@
+ 
+ # Ensure this gcc optimization is switched off for mips platforms to avoid trouble with JIT.
+ gcc:isEqual(QT_ARCH, "mips"): QMAKE_CXXFLAGS += -fno-reorder-blocks
++gcc:isEqual(QT_ARCH, "riscv64"): LIBS += -latomic
```
This differs form the mentioned patch by excluding the check for riscv32. This is ok as we do not release riscv32 binaries:

original patch's relevant line
```diff
+ gcc:if(isEqual(QT_ARCH, "riscv32")|isEqual(QT_ARCH, "riscv64")): LIBS += -latomic
```

jarolrod commented:
ACK 3a0f948c0bc93b571168165c54966bf164dc356a

Confirmed the issue on master, the supplied patches in this PR allow me to successfully build. I did not actually test the resulting binary on riscv.
